sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/SipRoutingClientBuilder.java

```diff
@@ -0,0 +1,451 @@
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Adds a {@link HttpPipelinePolicy pipeline policy} to apply on each request sent.
+     *
+     * <p><strong>Note:</strong> It is important to understand the precedence order of the HttpTrait APIs. In
+     * particular, if a {@link HttpPipeline} is specified, this takes precedence over all other APIs in the trait, and
+     * they will be ignored. If no {@link HttpPipeline} is specified, a HTTP pipeline will be constructed internally
+     * based on the settings provided to this trait. Additionally, there may be other APIs in types that implement this
+     * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
+     * documentation of types that implement this trait to understand the full set of implications.</p>
+     *
+     * @param policy A {@link HttpPipelinePolicy pipeline policy}.
+     * @return The updated {@link SipRoutingClientBuilder} object.
+     * @throws NullPointerException If {@code policy} is {@code null}.
+     */
+    @Override
+    public SipRoutingClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        this.additionalPolicies.add(Objects.requireNonNull(policy, "'policy' cannot be null."));
+        return this;
+    }
+
+    /**
+     * Allows for setting common properties such as application ID, headers, proxy configuration, etc. Note that it is
+     * recommended that this method be called with an instance of the {@link HttpClientOptions}
+     * class (a subclass of the {@link ClientOptions} base class). The HttpClientOptions subclass provides more
+     * configuration options suitable for HTTP clients, which is applicable for any class that implements this HttpTrait
+     * interface.
+     *
+     * <p><strong>Note:</strong> It is important to understand the precedence order of the HttpTrait APIs. In
+     * particular, if a {@link HttpPipeline} is specified, this takes precedence over all other APIs in the trait, and
+     * they will be ignored. If no {@link HttpPipeline} is specified, a HTTP pipeline will be constructed internally
+     * based on the settings provided to this trait. Additionally, there may be other APIs in types that implement this
+     * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
+     * documentation of types that implement this trait to understand the full set of implications.</p>
+     *
+     * @param clientOptions A configured instance of {@link HttpClientOptions}.
+     * @return The updated {@link SipRoutingClientBuilder} object.
+     * @throws NullPointerException If {@code clientOptions} is {@code null}.
+     * @see HttpClientOptions
+     */
+    @Override
+    public SipRoutingClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = Objects.requireNonNull(clientOptions, "'clientOptions' cannot be null.");
```

Comment:
`clientOptions` should be allowed to be null as they don't carry any required configuration